Index: fcgi-2.4.0/config.sub
===================================================================
--- fcgi-2.4.0.orig/config.sub	2014-07-02 14:09:51.812695076 -0400
+++ fcgi-2.4.0/config.sub	2014-07-02 14:10:32.240256866 -0400
@@ -259,6 +259,11 @@
 	i*86 | x86_64)
 	  basic_machine=$basic_machine-pc
 	  ;;
+
+	armeb-*)
+		basic_machine=$basic_machine-unknown
+		os=-linux--uclibcgnueabi
+		;;
 	# Object if more than one company name word.
 	*-*-*)
 		echo Invalid configuration \`$1\': machine \`$basic_machine\' not recognized 1>&2
